<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>advance_setting_dialog</class>
 <widget class="QDialog" name="advance_setting_dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1234</width>
    <height>590</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Advance setting</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Hash algorithms</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout">
             <item row="0" column="0">
              <widget class="QRadioButton" name="rb_avg_hash">
               <property name="text">
                <string>Average hash</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="hs_avg_hash">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="pb_avg_hash_default">
               <property name="text">
                <string>Default value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0" colspan="2">
              <widget class="QRadioButton" name="rb_bm_hash_0">
               <property name="text">
                <string>Block mean hash 0</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="hs_bmh_0">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>12</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox_2">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="minimum">
                <number>0</number>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>12</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="pb_bmh_0_default">
               <property name="text">
                <string>Default value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_5">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QRadioButton" name="rb_bm_hash_1">
               <property name="text">
                <string>Block mean hash 1</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="hs_bmh_1">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>48</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox_3">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>48</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="pb_bmh_1_default">
               <property name="text">
                <string>Default value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_6">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="0">
              <widget class="QRadioButton" name="rb_marr_hash">
               <property name="text">
                <string>Marr-Hildreth hash</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="hs_marr_hash">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>30</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox_4">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>30</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="pb_marr_hildreth_default">
               <property name="text">
                <string>Default value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_7">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0">
              <widget class="QRadioButton" name="rb_phash">
               <property name="text">
                <string>PHash</string>
               </property>
               <attribute name="buttonGroup">
                <string notr="true">buttonGroup</string>
               </attribute>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSlider" name="hs_phash">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QSpinBox" name="spinBox_5">
               <property name="toolTip">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="whatsThis">
                <string>The lower the threshold,the more similar the images pair</string>
               </property>
               <property name="maximum">
                <number>100</number>
               </property>
               <property name="value">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0" colspan="2">
              <widget class="QPushButton" name="pb_phash_default">
               <property name="text">
                <string>Default value</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Hash accuracy</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Orientation</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="combox_hash_orien"/>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="single_bar_plot" name="hash_accuracy_chart" native="true"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>single_bar_plot</class>
   <extends>QWidget</extends>
   <header>single_bar_plot.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>advance_setting_dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>advance_setting_dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>580</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_avg_hash</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>154</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_avg_hash</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>143</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_bmh_0</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>412</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>458</x>
     <y>147</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_bmh_0</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>455</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>433</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_bmh_1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_3</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>667</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>692</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_3</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_bmh_1</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>692</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>671</x>
     <y>152</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_marr_hash</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_4</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>915</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>931</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_4</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_marr_hash</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>931</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>912</x>
     <y>153</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hs_phash</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinBox_5</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1146</x>
     <y>140</y>
    </hint>
    <hint type="destinationlabel">
     <x>1170</x>
     <y>143</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinBox_5</sender>
   <signal>valueChanged(int)</signal>
   <receiver>hs_phash</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1179</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>1130</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
